### PR TITLE
schedule: fix a thread-safe bug and improve code (#1719)

### PR DESCRIPTION
**Pull request by Luffbee**
### What problem does this PR solve? 

cherry-pick #1719

### CODE REVIEW

**Comment by nolouch** [comment]
/run-all-tests tidb=release-2.1 tikv=release-2.1 tidb-test=release-2.1

**Comment by nolouch** [comment]
/run-all-tests tidb=release-3.0 tikv=release-3.0 tidb-test=release-3.0

File: server/schedule/operator_controller.go [hunk 1]
```diff
@@ -210,7 +219,7 @@ func (oc *OperatorController) AddOperator(ops ...*Operator) bool {
 
 	if oc.exceedStoreLimit(ops...) || !oc.checkAddOperator(ops...) {
 		for _, op := range ops {
-			operatorCounter.WithLabelValues(op.Desc(), "canceled").Inc()
+			operatorCounter.WithLabelValues(op.Desc(), "cancel").Inc()
```
Comment (rleungx) [inline server/schedule/operator_controller.go]: We also need to make a change in `tidb-ansible` for both `master` and `release-3.0` branches

File: server/schedule/operator_controller.go [hunk 2]
```diff
@@ -286,10 +295,10 @@ func (oc *OperatorController) addOperatorLocked(op *Operator) bool {
 	// If there is an old operator, replace it. The priority should be checked
 	// already.
 	if old, ok := oc.operators[regionID]; ok {
+		_ = oc.removeOperatorLocked(old)
 		log.Info("replace old operator", zap.Uint64("region-id", regionID), zap.Reflect("operator", old))
-		operatorCounter.WithLabelValues(old.Desc(), "replaced").Inc()
+		operatorCounter.WithLabelValues(old.Desc(), "replace").Inc()
```
Comment (rleungx) [inline server/schedule/operator_controller.go]: ditto

**Comment by Luffbee** [comment]
> After thinking about it, I think we don't need to update these two metrics. Because the people who upgrade PD to the next version maybe not upgrade tidb-ansible.

Changed them back.

**Pull request merged**